### ISSUE TITLE
Fix FP S109 ('no-magic-numbers'): Ignore numeric literal types

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#3401]
The rule S109 ('no-magic-numbers') reports false positives for numeric literal types:

```typescript
type Zero = 0;
type Five = 5;
```

To kill the noise, we need to enable the option [ignoreNumericLiteralTypes](https://typescript-eslint.io/rules/no-magic-numbers/#ignorenumericliteraltypes) from TypeScript ESLint's extension of the rule.

Reported in https://discuss.sonarsource.com/t/typescript-s109-possible-fp/11752
